rust/operator-binary/src/backend/dynamic.rs

```diff
@@ -86,3 +98,41 @@ pub async fn from_class(
         ),
     })
 }
+
+#[derive(Debug, Snafu)]
+#[snafu(module)]
+pub enum FromSelectorError {
+    #[snafu(display("failed to get {class}"))]
+    GetSecretClass {
+        source: stackable_operator::error::Error,
+        class: ObjectRef<SecretClass>,
+    },
+    #[snafu(display("failed to initialize backend for {class}"))]
+    FromClass {
+        source: FromClassError,
+        class: ObjectRef<SecretClass>,
+    },
+}
+
+impl SecretBackendError for FromSelectorError {
+    fn grpc_code(&self) -> tonic::Code {
+        match self {
+            FromSelectorError::GetSecretClass { .. } => tonic::Code::Unavailable,
```

Comment:
What about `tonic::Code::NotFound` ("Some requested entity was not found.")?
```suggestion
            FromSelectorError::GetSecretClass { .. } => tonic::Code::NotFound,
```

Reply:
So this is a bit more complicated here, since `Unavailable` _would_ be the correct return code for stuff like 500 or connection errors. It might still be worth adding separate handling here, but I'd put that in a new (lower priority) ticket.

Reply:
fine for me